File: space-game/Data.InMemoryData/DataProvider.vb
Imports Data
Imports Data.Classes

Public Class DataProvider
    Implements IDataProvider

    Private _solarSystem As List(Of CelestialObjectData)

    Public ReadOnly Property SolarSystem As List(Of CelestialObjectData) Implements IDataProvider.SolarSystem
        Get
            If _solarSystem Is Nothing Then
                _solarSystem = New SolarSystemData().SolSystem()
            End If
            Return _solarSystem
        End Get
    End Property
End Class

File: Data.ConfigDataProvider/Classes/DataProvider.cs
using Data.Classes;
using Jespers.Config;
using Mapster;
using System.Collections.Generic;

namespace Data.ConfigDataProvider.Classes
{
    public class DataProvider : IDataProvider
    {
        public List<CelestialObjectData> SolarSystem
        {
            get
            {
                SolarSystem system = AppConfig.Provider.Get<SolarSystem>();
                return system.Adapt<List<CelestialObjectData>>();
            }
        }
    }
}
